packages/drivers/debugger/src/fluidDebuggerController.ts

```diff
@@ -144,6 +149,24 @@ export class DebugReplayController extends ReplayController implements IDebugger
         reader.readAsText(file, "utf-8");
     }
 
+    public async onDownloadOps(): Promise<string> {
+        if (this.documentService !== undefined) {
+            const documentDeltaStorageService = await this.documentService.connectToDeltaStorage();
+            const messages = await this.fetchOpsFromDeltaStorage(documentDeltaStorageService);
+            return JSON.stringify(messages, undefined, 2);
+        }
+        return "N/A";
+    }
+
+    private async fetchOpsFromDeltaStorage(documentDeltaStorageService): Promise<ISequencedDocumentMessage[]> {
+        const deltaGenerator = generateSequencedMessagesFromDeltaStorage(documentDeltaStorageService);
+        let messages: ISequencedDocumentMessage[] = [];
+        for await (const message of deltaGenerator) {
```

Comment:
The last time I tries to use async generators in shipping code, I had to revert my changes because Fluid preview app was supporting some browser (Edge?) that did not support async generators.
Are you sure it's Ok? :)
That said, it's likely not critical here as this code is invoked only by developers

Reply:
Haha yes, we don't support old Edge anymore, so we should be good :)  (https://caniuse.com/?search=async%20generator)